pkg/geo/geomfn/linestring.go

```diff
@@ -252,3 +253,98 @@ func removePoint(lineString *geom.LineString, index int) (*geom.LineString, erro
 
 	return lineString.SetCoords(coords)
 }
+
+// LineSubString returns a LineString being a substring by start and end
+func LineSubString(g geo.Geometry, start, end float64) (geo.Geometry, error) {
+
+	if start < 0 || start > 1 || end < 0 || end > 1 {
+		return g, pgerror.Newf(pgcode.InvalidParameterValue,
+			"start and and must be within 0 and 1")
+	}
+	if start > end {
+		return g, pgerror.Newf(pgcode.Internal,
```

Comment:
can you make the code pgcode.InvalidParameterValue and error message "end must be greater or equal to the start"?